frontend/ui/widget/spinwidget.lua

```diff
@@ -338,4 +340,20 @@ function SpinWidget:onClose()
     return true
 end
 
+--[[
+This method updates the widget's value based on the direction of the spin.
+
+@param direction {number}. The direction of the spin (-1 for decrease, 1 for increase)
+@return boolean Always returns true to indicate the event was handled
+]]
+function SpinWidget:onSpinButtonPressed(direction)
+    local widget = self.value_widget
+    local step = self.value_step or 1
+    -- Use the widget's changeValue method but don't update the stored value directly
+    local new_value = widget:changeValue(widget:getValue(), direction * step, widget.value_max, widget.value_min, false)
```

Comment:
I have no real preference myself, but in NumberPickerWidget it says `self:changeValue(self.value, self.value_step * -1, self.value_max, self.value_min, self.wrap)`.

(On a semi-related note, that step = self.value_step or 1 seems a but unnecessary?)
```suggestion
    local new_value = widget:changeValue(widget:getValue(), step * direction, widget.value_max, widget.value_min, false)
```

Reply:
I think you might be correct, I was assuming the ones with value_table didn't have one

https://github.com/koreader/koreader/blob/a15c294bc2f645ba65f1fc4de6e2da537c11ef6a/frontend/ui/widget/numberpickerwidget.lua#L43

Reply:
both comments also apply to the other one btw, not just SpinWidget